src/reflect.rs

```diff
@@ -78,6 +78,11 @@ impl<'a> ReflectIntermediate<'a> {
             .unwrap_or(0)
             .into()
     }
+    fn get_var_component_or_default(&self, var_id: VariableId) -> Component {
+        self.get_deco_u32(var_id, None, Decoration::Component)
+            .unwrap_or(0)
+            .into()
+    }
```

Comment:
If we decided to merge the types then it might be a good idea to merge `get_var_location_or_default` and `get_var_component_or_default` too.